route/build/src/base_route.f90

```diff
@@ -0,0 +1,60 @@
+MODULE base_route
+
+  ! Abstract class for routing method
+
+  implicit none
+
+  private
+  public:: base_route_rch
+  public:: routeContainer
+
+  ! --- routing method container
+  type :: routeContainer
+    class(base_route_rch), allocatable :: rch_route
+  end type
+
+
+  ! --- routing method container
+  type, abstract :: base_route_rch
+    CONTAINS
+    procedure(sub_route_rch), deferred :: route
+  end type
+
+  ABSTRACT INTERFACE
+
```

Comment:
I suggest adding some comments to talk a bit about this subroutine and what's required in it.

Reply:
adding descriptions of the routine